File: scripts/Reverse Bits/Reverse Bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int mask = 0;
        int smask = 0;
        int j = 0;
        int rev = 0;
        
        // basically we are checking that the number is set bit or not 
        // if the number is set bit then we are appending that to our main answer i.e, rev
        for(int i=31 ; i>=0 ; i--){
            mask = 1<<i;
            if((mask&n)!=0){
                smask = 1<<j;
                rev = rev|smask;
            }
            j++;
        }
        
        // Time Complexity : O(32 for int)
        // Space Complexity : O(1)
        return rev;
    }
}

File: scripts/algorithms/R/Reverse Bits/Reverse Bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int ans =0;
        int i =1;
        int bit =0;
        while(i<32){
            bit = n&1;
            ans = ans|bit;
            n = n>>1;
            ans = ans<<1;
            i++;
        }
        if (i ==32){
            bit = n&1;
            ans = ans|bit;
        }
        return ans;
    }
};
